### ISSUE TITLE
How to use Batchnormalization effectly ?

### ISSUE THREAD

**Issue by Rpersie** [mravanelli/pytorch-kaldi#26]
Dear mravanelli,
  Recently, I have read your paper "BATCH-NORMALIZED JOINT TRAINING FOR DNN-BASED
DISTANT SPEECH RECOGNITION" .  In the paper it says the batchnormalization can replace the pretraining and can get significant improvement. 
   Now I want to use the joint training in the AMI dataset. But I can not get any performance when using     batchnormalization. Can you give me some idea of how you use the batchnorm. Or does the pytorch-kaldi  have the environment in your paper?  
Thank you !

**Comment by mravanelli** [comment]
Hi,
a short tutorial about joint training is under preparation. In general,
according to my experiments, batch normalization helps a lot when applied
to feedforward connections.

Mirco

**Comment by Rpersie** [comment]
Thank you for your reply ! I am looking forward to it.  In my experiment, the acmodel is lstm, featuremapping is DNN.  In your paper, all is dnn.  And the batchnorm is added only in featuremapping DNN. Do you think is this the problem?  Thank you !

**Comment by mravanelli** [comment]
I think you can apply batch norm to all the feedforward connections. You
can see an example of that into neural_networks.py (see class LSTM).

Mirco
